workbench-core/environments/src/services/environmentTypeConfigService.ts

```diff
@@ -28,6 +32,31 @@ export default class EnvironmentTypeConfigService {
     this._dynamoDbService = dynamoDbService;
   }
 
+  /**
+   * Soft Environment Type Configuration
+   * @param request - request for deleting environment type config
+   * @param checkDependency - check whether we can delete the envTypeConfig. The function should throw a Boom error if envTypeConfig cannot be deleted
+   * @returns void
+   */
+  public async softDeleteEnvironmentTypeConfig(
+    request: DeleteEnvironmentTypeConfigRequest,
+    checkDependency: (envTypeId: string, envTypeConfigId: string) => Promise<void>
+  ): Promise<void> {
+    const { envTypeId, envTypeConfigId } = request;
+    await checkDependency(envTypeId, envTypeConfigId);
+    await this.getEnvironmentTypeConfig(envTypeId, envTypeConfigId);
+
+    try {
+      await this._dynamoDbService.updateExecuteAndFormat({
+        key: this._buildEnvTypeConfigPkSk(envTypeConfigId),
+        params: {
+          item: { resourceType: `${this._resourceType}_deleted` }
+        }
+      });
+    } catch (e) {
```

Comment:
We should log error. I wonder why there were no eslint warnings regarding unused variable